tests/rptest/tests/schema_registry_test.py

```diff
@@ -1223,6 +1224,118 @@ def move_leader():
+                json.dumps({"schema": schema3_def})
+            ]
+            schema_ids = []
+
+            for schema in schemas:
+                self.logger.debug(f"Post schema {json.loads(schema)}")
+                result_raw = self._post_subjects_subject_versions(
+                    subject=subject, data=schema)
+                self.logger.debug(result_raw)
+                assert result_raw.status_code == requests.codes.ok
+                res = result_raw.json()
+                self.logger.debug(res)
+                assert "id" in res
+                schema_ids.append(result_raw.json()["id"])
+
+            return schemas, schema_ids
+
+        # Given a list of subjects, check that they exist on the registry
+        def check_subjects(subjects: list[str]):
+            result_raw = self._get_subjects()
+            self.logger.debug(result_raw)
+            assert result_raw.status_code == requests.codes.ok
+            res_subjects = result_raw.json()
+            self.logger.debug(res_subjects)
+            assert type(res_subjects) == type([])
+            res_subjects.sort()
+            subjects.sort()
+            assert res_subjects == subjects
+
+        # Given the subject and list of versions, check that the version numbers match
+        def check_subject_versions(subject: str, subject_versions: list[int]):
+            result_raw = self._get_subjects_subject_versions(subject=subject)
+            self.logger.debug(result_raw)
+            assert result_raw.status_code == requests.codes.ok
+            res_versions = result_raw.json()
+            self.logger.debug(res_versions)
+            assert type(res_versions) == type([])
+            res_versions.sort()
+            subject_versions.sort()
+            assert res_versions == subject_versions
+
+        # Given the subject, list of schemas, list of versions, and list of ids,
+        # check the schema that maps to the id
+        def check_each_schema(subject: str, schemas: list[str],
+                              schema_ids: list[int],
+                              subject_versions: list[int]):
+            for version in subject_versions:
+                self.logger.debug(
+                    f"Fetch schema version {version} on subject {subject}")
+                result_raw = self._get_subjects_subject_versions_version(
+                    subject=subject, version=version)
+                self.logger.debug(result_raw)
+                assert result_raw.status_code == requests.codes.ok
+                res = result_raw.json()
+                self.logger.debug(res)
+                assert type(res) == type({})
+                assert "subject" in res
+                assert "version" in res
+                assert "id" in res
+                assert "schema" in res
+                assert res["subject"] == subject
+                assert res["version"] == version
+                assert res["id"] in schema_ids
```

Comment:
The order matters here. The versions and their IDs are related. Perhaps `enumerate(subject_versions)` and index into `schema_ids`? Or `zip` them? This relies on not sorting the results earlier.

It might be better to change the representation:
map(subject,tuple<version, id>)

Reply:
done in https://github.com/redpanda-data/redpanda/pull/8127/commits/f0417002c0a8d5772987cb92f736fc3543f94375